bazaarvoice/emodb
# Emo 7119

## Github Issue #

`None`

## What Are We Doing Here?

Currently the megabus's Kafka producer uses static config as defined in `DefaultKafkaCluster.java`. The configuration of `linger.ms`, `batch.size`, and `buffer.memory` do not affect data correctness, but can greatly impact performance. This PR makes these configurable so that we can tinker and maximize throughput during boot.

## How to Test and Verify

1. Check out this PR
2. Configure the producer the way you want it in `config-megabus.yaml`
3. Run the megabus
4. check the logs for the producer configuration  that you set and ensure that it as intended 

## Risk

### Level 

`Low`

### Required Testing

`Manual`

### Risk Summary

The code changes here are very straightforward and contained. The only risk here is that there may be different optimal configurations for boot time and normal operating time, which may mean that we will have to remember to configure the megabus differently in certain situations.

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
